TrackingAntennaV3/main/Magnetometer.cpp

```diff
@@ -1,78 +1,66 @@
-/**
- * @file   Magnetometer.cpp
- * @Author Michael Lenover (WARG)
- * @date   January 6, 2017
- */
-
 #include "Logger.hpp"
 #include "Magnetometer.hpp"
 #include <Adafruit_LSM303_U.h>
+#include <MatrixMath.h> //Used for magnetometer calibration
 #include <MatrixMath.h>
 
+//Initializes magnetometer object
+static Adafruit_LSM303_Mag_Unified mag; 
 //Create magnetometer object
 Adafruit_LSM303_Mag_Unified mag = Adafruit_LSM303_Mag_Unified(COMPASS_ID);
 
 //Declare magnetometer event variable
+static sensors_event_t magEvent;
+
+//Initializes magnetometer struct
+volatile Magnetometer magnetometer;
 sensors_event_t magEvent;
 
 //Initialize magnetometer
 void initMagnetometer(){
+  //Defines mag object based on compass id
+  mag = Adafruit_LSM303_Mag_Unified(COMPASS_ID);
+
   
   //Enables auto-range
   mag.enableAutoRange(true);
   
+  //Initializes magnetometer
   //Initializes magnetomer
   if(!mag.begin()){
       //If initialization fails, send error message
-    	error("Compass failed to be detected.");
-  }
-  
-  else{
-    info("Magnetometer initialized.");
-  }
-}
+      error("Compass failed to be detected.");
```

Comment:
also for both the accelorometer, i'd maybe add an info message stating that the magnetometer was initized if successful